src/shared/Microsoft.Git.CredentialManager.Tests/WslUtilsTests.cs

```diff
@@ -0,0 +1,111 @@
+        [InlineData(@"\\wsl$", false)]
+        [InlineData(@"\\wsl$\", false)]
+        [InlineData(@"\\wsl$\ubuntu", true)]
+        [InlineData(@"\\wsl$\ubuntu\", true)]
+        [InlineData(@"\\wsl$\ubuntu\home", true)]
+        [InlineData(@"\\WSL$\UBUNTU\home", true)]
+        [InlineData(@"\\wsl$\ubuntu\home\", true)]
+        [InlineData(@"\\wsl$\openSUSE-42\home", true)]
+        public void WslUtils_IsWslPath(string path, bool expected)
+        {
+            bool actual = WslUtils.IsWslPath(path);
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData(@"\\wsl$\ubuntu", "ubuntu", "/")]
+        [InlineData(@"\\wsl$\ubuntu\", "ubuntu", "/")]
+        [InlineData(@"\\wsl$\ubuntu\home", "ubuntu", "/home")]
+        [InlineData(@"\\wsl$\ubuntu\HOME", "ubuntu", "/HOME")]
+        [InlineData(@"\\wsl$\UBUNTU\home", "UBUNTU", "/home")]
+        [InlineData(@"\\wsl$\ubuntu\home\", "ubuntu", "/home/")]
+        [InlineData(@"\\wsl$\openSUSE-42\home", "openSUSE-42", "/home")]
+        public void WslUtils_ConvertToDistroPath(string path, string expectedDistro, string expectedPath)
+        {
+            string actualPath = WslUtils.ConvertToDistroPath(path, out string actualDistro);
+            Assert.Equal(expectedPath, actualPath);
+            Assert.Equal(expectedDistro, actualDistro);
+        }
+
+        [PlatformTheory(Platforms.Windows)]
+        [InlineData(null)]
+        [InlineData(@"")]
+        [InlineData(@" ")]
+        [InlineData(@"\")]
+        [InlineData(@"wsl")]
+        [InlineData(@"\wsl\ubuntu\home")]
+        [InlineData(@"\\wsl\ubuntu\home")]
+        [InlineData(@"\wsl$\ubuntu\home")]
+        [InlineData(@"wsl$\ubuntu\home")]
+        [InlineData(@"//wsl$/ubuntu/home")]
+        [InlineData(@"\\wsl")]
+        [InlineData(@"\\wsl$")]
+        [InlineData(@"\\wsl$\")]
+        public void WslUtils_ConvertToDistroPath_Invalid_ThrowsException(string path)
+        {
+            Assert.Throws<ArgumentException>(() => WslUtils.ConvertToDistroPath(path, out _));
+        }
+
+        [PlatformFact(Platforms.Windows)]
+        public void WslUtils_CreateWslProcess()
+        {
+            const string distribution = "ubuntu";
+            const string command = "/usr/lib/git-core/git version";
+
+            string expectedFileName = WslUtils.GetWslPath();
+            string expectedArgs = $"--distribution {distribution} --exec {command}";
+
+            Process process = WslUtils.CreateWslProcess(distribution, command);
+
+            Assert.NotNull(process);
+            Assert.Equal(expectedArgs, process.StartInfo.Arguments);
+            Assert.Equal(expectedFileName, process.StartInfo.FileName);
+            Assert.True(process.StartInfo.RedirectStandardInput);
+            Assert.True(process.StartInfo.RedirectStandardOutput);
+            Assert.True(process.StartInfo.RedirectStandardError);
+            Assert.False(process.StartInfo.UseShellExecute);
+        }
+
```

Comment:
This might be overkill, but we might also test that `git --exec-path` returns the expected value.

Reply:
We cannot do this easily as we need to be invoked from a WSL distro, with specific environment variables set etc. These tests don't actually start any processes, just check that the `Process` object is constructed correctly, and is ready to be `.Start()`ed.